app/controllers/katello/api/v2/host_subscriptions_controller.rb

```diff
@@ -43,6 +44,40 @@ def destroy
       respond_for_destroy(:resource => @host)
     end
 
+    api :POST, "/hosts/subscriptions/", N_("Register a host with subscription and information.")
+    param :name, String, :desc => N_("Name of the host"), :required => true
+    param :uuid, String, :desc => N_("UUID to use for registered host")
+    param :facts, Hash, :desc => N_("Key-value hash of subscription-manager facts") do
+      param :fact, String, :desc => N_("Any number of sbscription facts about this host")
+    end
+    param :hypervisor_guest_uuids, Array, :desc => N_("UUIDs of the virtual guests from the host's hypervisor")
+    param :installed_products, Array, :desc => N_("List of products installed on the host")
```

Comment:
Snake case of what though? I think my point is being missed. We are saying this is a Katello supported API and it includes a parameter named "installed_products" and we have an entity that is called Products. However, these "installed_products" do not follow our conventions, and they refer to parameters that I find less than clear.

How does a user know to follow this format that includes specifying a product ID that is not the product ID? For me, this either needs to be not included in the docs or needs to be improved to clarify this to users or support how you would specify Katello products.